model/src/plugin_handler.cpp

```diff
@@ -773,16 +773,31 @@ bool PluginHandler::archive_check(int r, const char* msg, struct archive* a) {
   return r >= ARCHIVE_WARN;
 }
 
+/*
+ * Helper function to extract a tarball into platform-specific user directories.
+ *
+ *   @param path: Path to tarball.
+ *   @param filelist: On return contains a list of files installed.
+ *   @param metadata_path: If non-empty, location where to extract plugin metadata.xml file.
+ *   @param only_metadata: If true don't install any files, just extract
+ *                         the metadata.xml file.
+ *   @return true on success.
+ *           false if metadata_path != "" and tarball does not contain metadata.xml file.
+ */
 bool PluginHandler::explodeTarball(struct archive* src, struct archive* dest,
                                    std::string& filelist,
                                    const std::string& metadata_path,
                                    bool only_metadata) {
   struct archive_entry* entry = 0;
   pathmap_t pathmap = getInstallPaths();
+  bool status = metadata_path == "";
```

Comment:
"bool  status" is not a nice variable name. This variable reflects a state. The rule of thumb for booleans are names like "is_something", here perhaps "is_metadata_ok " or so.

Reply:
Agreed. I have renamed to `is_metadata_ok`.